google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/ui/DisablePluginWarningDialog.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.ui;
+
+import com.google.cloud.tools.intellij.util.GctBundle;
+
+import com.intellij.ide.plugins.IdeaPluginDescriptor;
+import com.intellij.ide.plugins.PluginManager;
+import com.intellij.ide.plugins.PluginManagerCore;
+import com.intellij.openapi.application.Application;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.application.ApplicationNamesInfo;
+import com.intellij.openapi.extensions.PluginId;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.openapi.util.SystemInfo;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.awt.Component;
+import java.awt.event.ActionEvent;
+
+import javax.swing.Action;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+
+/**
+ * A Dialog that prompts a user to disable a plugin.
+ */
+public class DisablePluginWarningDialog extends DialogWrapper {
+  private JLabel myPromptLabel;
+  private JLabel myRestartLabel;
+  private JPanel myContentPane;
+
+  private static final int DISABLE_EXIT_CODE = OK_EXIT_CODE;
+  private static final int DISABLE_AND_RESTART_EXIT_CODE = NEXT_USER_EXIT_CODE;
+  private final boolean myRestartCapable;
+
+  private DisablePluginWarningDialog(@NotNull Component parent, String pluginName,
+      boolean restartCapable) {
+    super(parent, false);
+    myRestartCapable = restartCapable;
+    myPromptLabel.setText(
+        GctBundle.message("error.dialog.disable.plugin.prompt", pluginName));
+    myRestartLabel
+        .setText(GctBundle.message(restartCapable
+            ? "error.dialog.disable.plugin.restart" : "error.dialog.disable.plugin.norestart",
+            ApplicationNamesInfo.getInstance().getFullProductName()));
+
+    setTitle(GctBundle.message("error.dialog.disable.plugin.title"));
+    init();
+  }
+
+  public static void disablePlugin(@NotNull PluginId pluginId, @NotNull Component parentComponent) {
```

Comment:
oh. I see what this is doing. It's a bit odd to call a static method on a dialog to perform a businessy action. I'll have to see where this is called from.